.github/scripts/summarize_tests.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# Set output file locations
+unit_test_file="./unit-test-output/unit_tests_output.txt"
+integration_test_file="./integration-test-output/integration_tests_output.txt"
+coverage_report_file="./coverage-output/coverage_report.json"
+
+# Initialize counters for unit tests
+unit_total_passed=0
+unit_total_failed=0
+unit_total_ignored=0
+unit_total_runtime=0.0
+
+# Initialize counters for integration tests
+integration_total_passed=0
+integration_total_failed=0
+integration_total_ignored=0
+integration_total_runtime=0.0
+
+overall_status="success"
+
+process_test_file() {
+  local file_type="$1" # "unit" or "integration"
+  local file_path="$2"
+  local output_not_found="$file_type test output not found."
+  
+  if [ -f "$file_path" ]; then
+    while IFS= read -r line; do
+      if [[ "$line" == *"test result:"* ]]; then
+        if [[ "$line" == *"failed; "* && "$overall_status" != "failed" ]]; then
+          overall_status="failed"
+        fi
+
+        passed=$(echo "$line" | awk '{for (i=1; i<=NF; i++) if ($i=="passed;") print $(i-1)}')
+        failed=$(echo "$line" | awk '{for (i=1; i<=NF; i++) if ($i=="failed;") print $(i-1)}')
+        ignored=$(echo "$line" | awk '{for (i=1; i<=NF; i++) if ($i=="ignored;") print $(i-1)}')
+        runtime=$(echo "$line" | awk -F'finished in ' '{n=split($2,a," "); print (n>0)?a[1]:0}')
```

Comment:
Use "" around $() to be safe in case returned value has special chars and/or whitespace(s)